resources/build/zip.inc.sh

```diff
@@ -23,13 +23,38 @@ function zip_files() {
   shift
 
   # Parse rest of parameters
-  FLAGS=()
-  INCLUDE=()
+  local ZIP_FLAGS=()
+  local SEVENZ_FLAGS=('a') # 7z requires a command
+  local INCLUDE=()
   while [[ $# -gt 0 ]] ; do
     case "$1" in
+      -r)
+        # Common flags to zip and 7z
+        ZIP_FLAGS+=($1)
+        SEVENZ_FLAGS+=($1)
+        shift
+        ;;
+      -q)
+        # quiet mode  -> disable progress indicator, set output log level 0
+        ZIP_FLAGS+=($1)
+        SEVENZ_FLAGS+=("-bd")
+        SEVENZ_FLAGS+=("-bb0")
+        shift
+        ;;
+      -[0123456789])
+        # Compression level where 
+        # -0 indicates no compression
+        # -1 indicates low compression (fastest)
+        # -9 indicates ultra compression (slowest)
+        ZIP_FLAGS+=($1)
+        if [[ $1 =~ -([0-9]) ]] then
+          SEVENZ_FLAGS+=("-mx${BASH_REMATCH[1]}")
+        fi  
+        shift;
+        ;;
       -*)
-        # zip/7z flags
-        FLAGS+=($1)
+        # Rest of zip flags.
+        ZIP_FLAGS+=($1)
```

Comment:
Don't we have to store the additional flags for 7zip as well? At least we should add a comment why we're not doing it.

Reply:
Only the zip flags that have a corresponding 7z flag are used (Handled in earlier cases). Will add comments